base/loading.jl

```diff
@@ -1593,8 +1593,31 @@ end
 """
     evalfile(path::AbstractString, args::Vector{String}=String[])
 
-Load the file using [`include`](@ref), evaluate all expressions,
-and return the value of the last one.
+Load the file into an anonymous module using [`include`](@ref), evaluate all expressions,
+and return the value of the last expression.
+The optional `args` argument can be used to set the input arguments of the script (i.e. the global `ARGS` variable).
+Note that definitions (e.g. methods, globals) are evaluated in the anonymous module and do not pollute the current module.
+
+# Example
+This example creates a temporary file (using [`mktemp`](@ref)) and writes a minimal julia script into it.
+The script is then executed via `evalfile` (with arguments) and prints the returned data.
+
```

Comment:
Do you think this is necessary? I think the example is pretty self-explanatory together with the explanation above.